File: Data Structures & Algorithms/length-of-last-word/submission-1.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        ptr_idx = len(s) - 1

        while ptr_idx >= 0:
            current = s[ptr_idx]
            if current == " ":
                ptr_idx -= 1
            else:
                break

        last_word_length = 0
        while ptr_idx >= 0 and s[ptr_idx] != " ":
            ptr_idx -= 1
            last_word_length += 1

        return last_word_length

File: Data Structures & Algorithms/valid-sudoku/submission-1.kt
class Solution {
    fun isValidSudoku(board: Array<CharArray>): Boolean {
        val rows = Array(9) { mutableSetOf<Char>() }
        val cols = Array(9) { mutableSetOf<Char>() }
        val squares = Array(9) { mutableSetOf<Char>() }

        for (row in 0 until 9) {
            for (col in 0 until 9) {
                val squareHash = (row / 3) * 3 + (col / 3)
                val current = board[row][col]

                if (current == '.') continue
                if (current in rows[row] || current in cols[col] || current in squares[squareHash]) {
                    return false
                }

                rows[row].add(current)
                cols[col].add(current)
                squares[squareHash].add(current)
            }
        }

        return true;
    }
}
